### ISSUE TITLE
`_lib._util.MapWrapper` uses multiprocessing with `fork`, deprecated in Python 3.12

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#19848]
Test failure from the new CI job with Python 3.12 I was just working on:

```
 ___________ TestDifferentialEvolutionSolver.test_immediate_updating ____________
[gw1] linux -- Python 3.12.1 /opt/hostedtoolcache/Python/3.12.1/x64/bin/python
scipy/optimize/tests/test__differential_evolution.py:703: in test_immediate_updating
    with warns(UserWarning):
E   DeprecationWarning: This process (pid=12452) is multi-threaded, use of fork() may lead to deadlocks in the child.
        bounds     = [(0.0, 2.0), (0.0, 2.0)]
        self       = <scipy.optimize.tests.test__differential_evolution.TestDifferentialEvolutionSolver object at 0x7f60117f4e30>
        solver     = <scipy.optimize._differentialevolution.DifferentialEvolutionSolver object at 0x7f5ffc3b0e30>

...

 FAILED scipy/optimize/tests/test__differential_evolution.py::TestDifferentialEvolutionSolver::test_immediate_updating - DeprecationWarning: This process (pid=12452) is multi-threaded, use of fork() may lead to deadlocks in the child.
```

This was inside a `pytest.warns` that was filtering another warning, so this is probably being surfaced now because of the combination of:
- the deprecation which is new in Python 3.12
- use of `pytest` 8.0, which bubbles up more warnings
- running the tests in parallel

Note that the warning isn't deterministic, because Python cannot reliably determine whether a program is using threading. I cannot reproduce it locally, but the cause is clear. See the notes on using the `fork` method with `multiprocessing` in https://docs.python.org/3/whatsnew/3.12.html.

For now I'll just avoid the warning, but I think the thing to do here is to switch the start method to `'spawn'` or `'forkserver'` (which one is TBD - the latter is faster, but not always available).

Cc @andyfaff

**Comment by andyfaff** [comment]
The `DeprecationWarning` message isn't very clear, it should be pointing to info like the release notes for cp312 give. i.e. `fork` is deprecated in 3.12 (for linux), and will be moving to something else in 3.14. If you don't want to filter the deprecation warning in the test one you'll have to provide your own `workers`:

```python
ctx = multiprocessing.get_context('spawn')
with ctx.Pool() as p:
    with warns(UserWarning):
        with DifferentialEvolutionSolver(rosen, bounds, workers=p.map) as solver:
                pass
        assert solver._updating == 'deferred'
```

Note, the `pytest.warns` context manager is checking that a `UserWarning` is emitted, it's not filtering it per-se.
